src/Nox.Generator/Nox.Generator.cs

```diff
@@ -205,4 +207,11 @@ private void GenerateSource(SourceProductionContext context, ImmutableArray<(str
 
         return true;
     }
+
+    private static string? GetProjectRootDirectory(ImmutableArray<(string Path, SourceText? Source)> noxYamls)
+    {
+        var generatorName = "\\generator.nox.yaml";
+        var generatorPath = noxYamls.FirstOrDefault(x => x.Path.EndsWith(generatorName)).Path;
+        return generatorPath?.Remove(generatorPath.Length - generatorName.Length);
```

Comment:
see the Path utility and for example https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getpathroot?view=net-7.0